frb_rust/src/handler.rs

```diff
@@ -252,23 +259,24 @@ impl<EH: ErrorHandler> Executor for ThreadPoolExecutor<EH> {
         });
     }
 
-    fn execute_sync<SyncTaskFn>(
+    fn execute_sync<SyncTaskFn, Er>(
         &self,
         _wrap_info: WrapInfo,
         sync_task: SyncTaskFn,
-    ) -> Result<SyncReturn<Vec<u8>>>
+    ) -> Result<SyncReturn<Vec<u8>>, Er>
     where
-        SyncTaskFn: FnOnce() -> Result<SyncReturn<Vec<u8>>> + UnwindSafe,
+        SyncTaskFn: FnOnce() -> Result<SyncReturn<Vec<u8>>, Er> + UnwindSafe,
+        Er: IntoDart,
     {
         sync_task()
     }
 }
 
 /// Errors that occur from normal code execution.
-#[derive(Debug)]
+//#[derive(Debug)]
 pub enum Error {
-    /// Errors from an [anyhow::Error].
-    ResultError(anyhow::Error),
+    /// Errors that implement [IntoDart].
+    CustomError(Box<dyn FnOnce() -> DartCObject>),
```

Comment:
Hmm shall we make it sth like `Box<dyn IntoDart>`

Reply:
No cause `IntoDart` consumes the `self`

Reply:
Well I mean a Box of an object that IntoDart. Memory-wise it should work. So. what error does it have?

Reply:
I tried `Box<dyn IntoDart>` but to call `.into_dart()` on that it has to consume itself, but the thing is inside a `Box`

Reply:
Then maybe consume the Box?

Reply:
I see the problem https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=523285a03d58a166a742587aafdfeea5

Let me think about it

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=057cccd0fcae7d98eb5b2d967584724c

```rust
trait IntoDart {
    fn into_dart(self) -> String;
}

struct A(String);
impl IntoDart for A {
    fn into_dart(self) -> String { self.0 }
}

trait BoxIntoDart {
    fn box_into_dart(self: Box<Self>) -> String;
}

impl<T: IntoDart> BoxIntoDart for T {
    fn box_into_dart(self: Box<T>) -> String {
        self.into_dart()
    }
}

fn main() {
    let a: Box<dyn BoxIntoDart> = Box::new(A("hello".to_string()));
    let b: String = a.box_into_dart();
    println!("b={}", b);
}

Reply:
ref： https://stackoverflow.com/questions/50949379/how-to-move-a-value-out-of-an-object-safe-trait-object

Reply:
In other words, we need to write down these code in our package:

```rust
trait BoxIntoDart {
    fn box_into_dart(self: Box<Self>) -> String;
}

impl<T: IntoDart> BoxIntoDart for T {
    fn box_into_dart(self: Box<T>) -> String {
        self.into_dart()
    }
}
```

and then you can use `Box<dyn BoxIntoDart>` freely :)